src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -315,17 +316,12 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		});
 	}
 
-	private _getIcon(): Codicon | undefined {
-		if (this.shellLaunchConfig.icon) {
-			return iconRegistry.get(this.shellLaunchConfig.icon);
+	private _getIcon(): Codicon | ThemeIcon | URI | { light: URI, dark: URI } | undefined | string {
+		const iconPath = this._shellLaunchConfig.iconPath || this._shellLaunchConfig.attachPersistentProcess?.icon;
```

Comment:
We no longer get string icons via the registry here, it would be simpler if we could get rid of the possibility of the `string` type as early as possible to simplify the type we're passing around